chat_with_nerf/visual_grounder/picture_taker.py

```diff
@@ -154,11 +167,33 @@ def take_picture(self, query: str, session_id: str) -> tuple[list[ImageRef], str
                 ]  # Get all members of the cluster
                 centroids.append(members.mean(axis=0))  # Compute centroid
 
-        # centroids -> camera poses
+                # Find the top value and corresponding point in the cluster
+                member_values = top_values[labels == cluster_id]
+                max_value_index = np.argmax(member_values)
+                max_index.append(max_value_index)
+                top_values_in_clusters.append(member_values[max_value_index])
+                top_points_in_clusters.append(members[max_value_index])
+
+        # Remove duplicate centroids by looking for its corresponding origins
+        top_origins_list = top_origins[max_index, :]
+        unique_position_dict = {}
+        for i, array in enumerate(top_origins_list):
+            tuple_array = tuple(array)
+            if tuple_array not in unique_position_dict:
+                unique_position_dict[tuple_array] = i
```

Comment:
should we really do this post-processing on the centroids? what if two centroids share the same origin but different directions? Then in this case the second centroid will be removed. However, the removing logic depends on the ordering of the centroids in the list, which is arbitrary.